internal/parser/scanner/states.go

```diff
@@ -357,230 +570,526 @@ var keywordsWithW map[string]token.Type = map[string]token.Type{
 // }
 
 func scanAKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithA[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithA[input], input, s)
+	nextRune := s.seekNext(s.start)
+	input := s.input[s.start:nextRune]
+	for _, k := range keywordsWithA {
+		keyword := strings.Split(k, "")
```

Comment:
why do you split `k`?
use `keyword := []rune(k)` instead

Reply:
Alright.